src/CRVBadDebtRepayment.sol

```diff
@@ -0,0 +1,121 @@
+
+    /// CRV cap fulfilled
+    error ExcessCRVAmountIn(uint256 amountLeft);
+    /// AUSDC cap fulfilled
+    error ExcessAUSDCPurchased(uint256 amountLeft);
+    /// Not enough CRV in contract to repay bad debt
+    error NotEnoughCRV(uint256 amount);
+    /// Oracle price is 0 or lower
+    error InvalidOracleAnswer();
+    /// Need to request more than 0 tokens out
+    error OnlyNonZeroAmount();
+
+    /// @notice Purchase USDC for CRV
+    /// @param amountIn Amount of BAL input
+    /// @param toUnderlying Whether to receive as USDC (true) or aUSDC (false)
+    /// @return amountOut Amount of USDC received
+    /// @dev Purchaser has to approve BAL transfer before calling this function
+    function purchase(uint256 amountIn, bool toUnderlying) external returns (uint256 amountOut) {
+        if (amountIn == 0) revert OnlyNonZeroAmount();
+        if (amountIn > availableCRVToBeFilled()) revert ExcessCRVAmountIn(availableCRVToBeFilled());
+
+        amountOut = getAmountOut(amountIn);
+        if (amountOut == 0) revert OnlyNonZeroAmount();
+        if (amountOut > availableAUSDCToBeSold()) revert ExcessAUSDCPurchased(availableAUSDCToBeSold());
+
+        totalCRVReceived += amountIn;
+        totalAUSDCSold += amountOut;
+
+        CRV.safeTransferFrom(msg.sender, address(this), amountIn);
+        if (toUnderlying) {
+            AUSDC.safeTransferFrom(AaveV2Ethereum.COLLECTOR, address(this), amountOut);
+            // Withdrawing entire aUSDC balance in this contract since we can't directly use 'amountOut' as
+            // input due to +1/-1 precision issues caused by rounding on aTokens while it's being transferred.
+            amountOut = AaveV2Ethereum.POOL.withdraw(address(USDC), type(uint256).max, msg.sender);
+            emit Purchase(address(CRV), address(USDC), amountIn, amountOut);
+        } else {
+            AUSDC.safeTransferFrom(AaveV2Ethereum.COLLECTOR, msg.sender, amountOut);
+            emit Purchase(address(CRV), address(AUSDC), amountIn, amountOut);
+        }
+
+        return amountOut;
+    }
+
+    /// @notice Returns how close to the 2,656,500 CRV amount cap we are
+    /// @return availableCRVToBeFilled the amount of BAL left to be filled
+    /// @dev Purchaser check this function before calling purchase() to see if there is CRV left to be filled
+    function availableCRVToBeFilled() public view returns (uint256) {
+        return CRV_CAP - totalCRVReceived;
+    }
+
+    /// @notice Returns how close to the 2,000,000 aUSDC amount cap we are
+    /// @return availableAUSDCToBeSold the amount of aUSDC left to be sold
+    /// @dev Purchaser check this function before calling purchase() to see if there is aUSDC left to be sold
+    function availableAUSDCToBeSold() public view returns (uint256) {
+        return AUSD_CAP - totalAUSDCSold;
+    }
+
+    function getAmountOut(uint256 amountIn) public view returns (uint256 amountOut) {
```

Comment:
This is going to return `amountOut` right? But the actual return is `(amountOut * 10010) / 10000`. I think it's confusing. 

Might be worth it to just have it as `uint256` here instead of `uint256 amountOut` since there's a return already at the end of the function.